docs/designs/schema-etl.md

```diff
@@ -0,0 +1,182 @@
+                items:
+                  properties:
+                    name:
+                      type: string
+                    type:
+                      type: string
+                  required:
+                  - name
+                  - type
+                  type: object
+                type: array
+              table:
+                description: |-
+                  Table is the name of the table in which the function will be created. We must
+                  specify a table if the function is a view, otherwise the Table name is optional.
+                type: string
+            required:
+            - body
+            - database
+            - name
+            type: object
+          status:
+            description: FunctionStatus defines the observed state of Function
+            properties:
+              lastTimeReconciled:
+                description: LastTimeReconciled is the last time the Function was
+                  reconciled
+                format: date-time
+                type: string
+              message:
+                description: Message is a human-readable message indicating details
+                  about the Function
+                type: string
+            required:
+            - lastTimeReconciled
+            - message
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+```
+
+A sample use is:
+
+```yaml
+apiVersion: adx-mon.azure.com/v1
+kind: Function
+metadata:
+  name: samplefn
+spec:
+  name: SampleFn
+  body: |
+    SampleFn
+    | extend Timestamp = todatetime(Body['ts']),
+            Message = tostring(Body['msg']),
+            Labels = Attributes['labels'],
+            Host = tostring(Resource['Host'])
+  database: SomeDatabase
+  table: SampleFn
+  isView: true
+```
+
+_Ingestor_ would then execute the following 
+
+```kql
+.create-or-alter function with ( view=true ) SampleFn () {
+    SampleFn
+    | extend Timestamp = todatetime(Body['ts']),
+            Message = tostring(Body['msg']),
+            Labels = Attributes['labels'],
+            Host = tostring(Resource['Host'])
+}
+```
+
+## Implementation Details
+
+- Ingestor already manages the creation of Tables as logs flow through the system, so we'll have Ingestor also manage creation of these Functions because the Table must exist prior to the Function that it references.
+- Alerter already handles CRDs, as much as possible we'll share code between these components so that we're not duplicating a bunch of code.
+- When attempting to reconcile each Function's state, we'll make sure the process is idempotent such that if a Function definition already exists that matches all the parameters found in its CRD, we do not attempt to update its state.
+
+## Questions
+- If Ingestor has a reference to a CRD / Function that is later deleted, do we want to delete the associated Function in Kusto? 
```

Comment:
I would lean to delete if the CRD is no longer present since it is easy to recreate, but open to other thoughts too.